### PR TITLE
finalise schema and auth for end of sprint #1

### PR DESCRIPTION
**Pull request by nathanSamosa**

### CODE REVIEW

**Comment by nathanSamosa** [comment]
For the schema to update in all the right places, empty your SQL database with `npx prisma migrate reset`. Since seed.js is empty this will just clear your database, and occupy it with nothing.

Then run `npx prisma generate` and `npx prisma migrate dev --create-only --skip-seed --name new_model` to reformat/create the migration needed to send data to SQL.

Finally, run  `npx prisma migrate reset` one last time to apply the new migration to your SQL database.

Now you can `npm start` and send data to your SQL using the routers.

File: prisma/schema.prisma [hunk 1]
```diff
@@ -15,14 +15,12 @@ model Pokemon {
   id             Int              @id @default(autoincrement())
   name           String
   art            String
-  type1          String
-  type2          String?
-  ratingAvg      Int              @default(1)
-  ratingCount    Int              @default(0)
-  createdAt      DateTime         @default(now())
-  updatedAt      DateTime         @updatedAt
+  typePrimary    String
```
Comment (TheunisK) [inline prisma/schema.prisma:18]: Nice. I think this makes much more sense.

File: src/client/App.jsx [hunk 2]
```diff
@@ -1,11 +1,13 @@
 import './App.css';
 
 import Login from './components/Login';
+import Register from './components/Registration';
 
 export default function App() {
   return (
     <div className="App">
       <Login />
+      <Register />
```
Comment (TheunisK) [inline src/client/App.jsx:10]: Has the component now been changed to Register? That works for me.

File: src/client/components/Login.jsx [hunk 3]
```diff
@@ -27,7 +27,7 @@ const Login = () => {
     const handleChange = e => {
         const { value, name } = e.target;
         setLoginDetails({
-            ...userDetails, [name]: value
+            ...loginDetails, [name]: value
```
Comment (TheunisK) [inline src/client/components/Login.jsx:30]: Thanks. I missed this one.

File: src/server/controllers/user.js [hunk 4]
```diff
@@ -34,16 +37,17 @@ const postLoginDetails = async (req, res) => {
...
-            }
-        } else {
-            return res.json("Username or Password is incorrect.");
+
+        const matchingPassword = await bcrypt.compare(password, findUser.password);
+
+        if (findUser && matchingPassword) {
+            const token = jwt.sign(username, secret);
+            console.log(token)
+            return res.json({ data: token })
```
Comment (TheunisK) [inline src/server/controllers/user.js:46]: This works so much better. Nice work. We should put half a day aside next week to test all the possible errors the user can make and make sure were handling them properly.

File: src/client/App.jsx (hunk 2 above)
Comment (nathanSamosa) [reply src/client/App.jsx:10]: Missed that, will change Register to Registration to match the exported component

File: src/server/controllers/user.js (hunk 4 above)
Comment (nathanSamosa) [reply src/server/controllers/user.js:46]: Agreed, currently there is a json error when defining matchingPassword if findUser is undefined, so need to make defining matchingPassword conditional.

File: src/client/components/Login.jsx (hunk 3 above)
Comment (Can-Tina) [reply src/client/components/Login.jsx:30]: Nice spot, Nathan

File: src/server/controllers/pokemon.js [hunk 5]
```diff
@@ -6,14 +6,15 @@ const pokemonNumber = 151;
...
     pokeArr.push({
         name: pokemon.species.name,
         art: pokemon.sprites.other.dream_world.front_default,
-        type1: pokemon.types[0].type.name
+        typePrimary: pokemon.types[0].type.name
     })
 
     if(pokemon.types[1]) {
-        pokeArr[arrIndex] = {...pokeArr[arrIndex], type2: pokemon.types[1].type.name}
+        pokeArr[arrIndex] = {...pokeArr[arrIndex], typeSecondary: pokemon.types[1].type.name}
```
Comment (Can-Tina) [inline src/server/controllers/pokemon.js:17]: I do like the primary and secondary more

File: src/server/controllers/user.js (hunk 4 above)
Comment (Can-Tina) [reply src/server/controllers/user.js:46]: Sounds like a plan!

**Comment by glowkeeper** [comment]
Approved

**Pull request merged**